From 0f66d6bc4f6db7401ccde33b342d6a6542b4da6b Mon Sep 17 00:00:00 2001
From: "Simon A. F. Lund" <os@safl.dk>
Date: Wed, 24 Nov 2021 14:48:51 +0100
Subject: [PATCH] eal: disable dynamic-loading

Signed-off-by: Simon A. F. Lund <os@safl.dk>
---
 lib/eal/common/eal_common_options.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/eal/common/eal_common_options.c b/lib/eal/common/eal_common_options.c
index ff5861b5f3..603f6f7413 100644
--- a/lib/eal/common/eal_common_options.c
+++ b/lib/eal/common/eal_common_options.c
@@ -530,16 +530,19 @@ is_shared_build(void)
 
 	RTE_LOG(INFO, EAL, "Detected static linkage of DPDK\n");
 	return 0;
 }
 
 int
 eal_plugins_init(void)
 {
+	// NOTE: disabling support for plugins to avoid linker issue with 'dl'
+	return 0;
+
 	struct shared_driver *solib = NULL;
 	struct stat sb;
 
 	/* If we are not statically linked, add default driver loading
 	 * path if it exists as a directory.
 	 * (Using dlopen with NOLOAD flag on EAL, will return NULL if the EAL
 	 * shared library is not already loaded i.e. it's statically linked.)
 	 */
-- 
2.30.2

