diff --git i/lib/librte_eal/common/eal_common_options.c w/lib/librte_eal/common/eal_common_options.c
index a5426e123..f73782904 100644
--- i/lib/librte_eal/common/eal_common_options.c
+++ w/lib/librte_eal/common/eal_common_options.c
@@ -484,6 +484,9 @@ eal_dlopen(const char *pathname)
 int
 eal_plugins_init(void)
 {
+	// NOTE: disabling support for plugins to avoid linker issue with 'dl'
+	return 0;
+
 	struct shared_driver *solib = NULL;
 	struct stat sb;
 
