From 93cbe5a2352ea6c62eabf8f64fce43fdd6b8b426 Mon Sep 17 00:00:00 2001
From: "Simon A. F. Lund" <os@safl.dk>
Date: Wed, 24 Nov 2021 14:48:51 +0100
Subject: [PATCH 2/2] eal: disable dynamic-loading

Signed-off-by: Simon A. F. Lund <os@safl.dk>
---
 lib/eal/common/eal_common_options.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/eal/common/eal_common_options.c b/lib/eal/common/eal_common_options.c
index 4c2def0155..0f0fa22b0b 100644
--- a/lib/eal/common/eal_common_options.c
+++ b/lib/eal/common/eal_common_options.c
@@ -530,6 +530,9 @@ is_shared_build(void)
 int
 eal_plugins_init(void)
 {
+	// NOTE: disabling support for plugins to avoid linker issue with 'dl'
+	return 0;
+
 	struct shared_driver *solib = NULL;
 	struct stat sb;
 
-- 
2.37.3

